File: app/src/main/scala/com/example/scala_3_android_java/Bar.scala
package com.example.scala_3_android_java

/**
 * A simple Scala object accessible from Java code
 */
object Bar:
  // A value that will be accessible from Java
  val foo: String = "Hello from Scala in app module!"
  
  // A method that returns the foo value
  def foo(): String = foo
  
  // Access the core module's Foo.bar value
  def getValueFromCore(): Int = com.example.core.Foo.bar 

File: app/src/main/java/com/example/scala_3_android_java/Bar.java
package com.example.scala_3_android_java;

import com.example.core.Foo;

public class Bar {
    // Value that matches the Scala impl
    private static final String FOO_VALUE = "Hello from Scala in app module!";
    
    // Getter method for the foo value
    public static String foo() {
        return FOO_VALUE;
    }
    
    // Method that calls the core module's Foo.bar method
    public static int getValueFromCore() {
        return Foo.bar();
    }
}
